Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif

```diff
@@ -0,0 +1,139 @@
+
+[MODEL: dataset=ibm_security_verify_raw]
+/*
+Mapping of 'management' event types.
+*/
+filter event_type = "management"
+| alter
+    src_user1 = arrayindex(regextract(data -> performedby_username, "(.+)\@") ,0), // The username can sometimes appear as a username and other times as an email.
+    src_user2 = arrayindex(regextract(data -> username, "(.+)\@") ,0),
+    mod = if(data -> modified = "[]", null, data -> modified)
+| alter
+    xdm.event.operation_sub_type = coalesce(data -> dict_op, data -> action),
+    xdm.target.application.name = data -> applicationname,
+    xdm.event.description = if(data -> cause != null AND data -> modified != null, concat(rtrim(data -> cause, "."), ". modified properties: ", mod), coalesce(data -> cause, data -> modified)), // Checks if 'cause' and 'modified' are both populated. If so, appends the cause with a list of modified properties. If not, returns whichever field is available.
+    xdm.event.original_event_type = data -> messageid,
+    xdm.auth.mfa.method = data -> mfamethod,
+    xdm.source.user.identifier = if(data -> performedby IN("System", "shared_apikey"), null, data -> performedby),
+    xdm.source.application.name = coalesce(data -> performedby_clientname, if(data -> performedby IN("System", "shared_apikey"), data -> performedby)),
+    xdm.source.user.username = coalesce(src_user2, data -> username, src_user1, data -> performedby_username), //2 possible username fields, performedby_username and username.
+    xdm.source.user.upn = if(src_user2 != null, data -> username ,src_user1 != null, data -> performedby_username),
+    xdm.target.resource.id = data -> entitlementid,
+    xdm.target.resource.name = data -> entitlementname,
+    xdm.target.resource.sub_type = data -> entitlementsubtype,
+    xdm.target.resource.type = if(data -> entitlementtype != null AND data -> entitlementcategory != null, concat(data -> entitlementtype, " - ", data -> entitlementcategory), coalesce(data -> entitlementtype, data -> entitlementcategory, data -> resource )), // The "type" field represents the role, while the "category" field specifies the role type, e.g., "role - admin."
+    xdm.source.user.groups = to_json_string(data -> entitlements) -> []
+| call generic_fields_mapping;
+
+/*
+Mapping of 'sso' event types.
+*/
+filter event_type = "sso"
+| alter
+    src_user = arrayindex(regextract(data -> username, "(.+)\@") ,0) // The username can sometimes appear as a username and other times as an email.
+| alter
+    xdm.auth.service = data -> subtype,
+    xdm.source.user.identifier = data -> userid,
+    xdm.source.user.username = coalesce(src_user, data -> username),
+    xdm.source.user.upn = if(src_user != null, data -> username),
+    xdm.target.application.name = data -> applicationname,
+    xdm.source.host.device_id = data -> deviceid //valid only if its an Android or iPhone device.
+| call generic_fields_mapping;
+
+/*
+Mapping of 'authentication' event types
```

Comment:
The following fields are missing:

xdm_target_resource_id (data.applicationid)
xdm.source.user.username (similar to DUO https://github.com/demisto/content/pull/29139/files)
xdm.source.user.first.name (similar to DUO)
xdm.source.user.last.name (similar to DUO)
xdm.event.original_event_type = event_type

Reply:
In 'authentication' events there isn't `data.applcationid` / `data.applicationname`
`xdm.source.user.username `- is being extracted and populated. 
For `xdm.source.user.first.name`and `xdm.source.user.last.name`, based on the DUO documentation, the logs contain fields like '[sam.smith@email.com](mailto:sam.smith@email.com)', so It makes sense to extract first and last names, but in IBM we're receiving UPN or username.
In `xdm.event.original_event_type`, it's currently populated with messageid, a unique ID number for the message in data.cause. This ID refers to a specific event subtype in an authentication event.

Reply:
@yassorpanw As discussed, we can resolve this conversation:

1. xdm.target.resource.id - No application fields in authentication events.
2. xdm.source.user.username - Added
3. xdm.source.user.first.name - Can’t be achieved (username can be anything)
4. xdm.source.user.last.name - Can’t be achieved (username can be anything)
5. xdm.event.original_event_type - set to event_type to help with tracing back the event.
6. xdm.event.operation - Added (if(lowercase(data.subtype) = "MFA",  XDM_CONST.OPERATION_TYPE_AUTH_MFA, XDM_CONST.OPERATION_TYPE_AUTH_LOGIN)